From 09c1ad24296b41b5a326ee900db64c59722bf1b7 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Mon, 9 Feb 2015 15:10:00 +0800
Subject: [PATCH 263/269] atomisp2:CHT:revise flow of config capture input

kernel panic happen when h_ds_factor != v_ds_factor,
and CTS case testYuvAndJpeg failed!
this is regression of Tuukka's patch:
https://icggerrit.ir.intel.com/#/c/36004
so do original flow for CHT

Tracked-On: IMINAN-12474
Change-Id: I8949c06b76015b5a01860c1f97431595071675ac
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/40209
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c        | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 09ba331..c7587db 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2634,9 +2634,9 @@ static void __configure_capture_pp_input(struct atomisp_sub_device *asd,
 	ver_ds_factor = CEIL_DIV(height >> 1,
 			pipe_configs->output_info[0].res.height);
 
-	if (asd->isp->media_dev.hw_revision <
-	    (ATOMISP_HW_REVISION_ISP2401 << ATOMISP_HW_REVISION_SHIFT) &&
-	    hor_ds_factor != ver_ds_factor) {
+	if ((asd->isp->media_dev.hw_revision <
+	    (ATOMISP_HW_REVISION_ISP2401 << ATOMISP_HW_REVISION_SHIFT) ||
+	    IS_CHT) && hor_ds_factor != ver_ds_factor) {
 		dev_warn(asd->isp->dev,
 				"Cropping for capture due to FW limitation");
 		return;
-- 
1.9.1

